docs/smart-contracts/guides/events/consume.mdx

```diff
@@ -1,7 +1,251 @@
 ---
 title: Consume previously ingested events
-hide_table_of_contents: true
-draft: true
 ---
 
-Placeholder text goes here.
+Once events have been ingested into a database, for instance as done in the [ingest guide], they can be consumed without having the need to query again Soroban RPC. In the following, we will show how we can consume these events in Python.
+
+To access the database, we will use [SQLAlchemy](https://www.sqlalchemy.org), which is one of the most used library used in Python to query database. Similarly to Prisma, SQLAlchemy is an Object Relational Mapper (ORM), allowing us to write database query directly in Python.
+
+Then, we will use [stellar-sdk] to decode the events and process them.
+
+Let's get started!
+
+## Setup
+
+In a virtual environment, install the Python dependencies:
```

Comment:
Ok yes, I will add that 👍 I was not sure how deep I had to cover that part since any guide showing some Python would need that. i.e. in the end I think it would be good to have a general setup page for the guides. This way we can present setups for all the languages on the same page and then assume all is good in the following sections.